### DISCUSSION TITLE
Why are the classes "dict-like" ?

### DISCUSSION THREAD

**Discussion by Jota-Ve** [cinemagoer/cinemagoer#377] [General]
So, this might be a question with an obvious answer, but honestly I don't get it. Why make it so the only way to access variables is through string keys ? 

Is there any benefit for structuring it like:
```
directors = movie['directors']
```

Instead of:
```
directors = movie.directors
```

Since the first doesn't offer code completion or type hints, it is way more likely to produce errors, so I do wonder why it was chosen to be this way ? And is there any plans to add the second option ?

**Comment by alberanid** [comment]
mostly it's because not all keys are known beforehand: some are taken by the title of the various pages and sections of information. Plus, some of them may have spaces in them.

It should be possible to support both dynamically, but I doubt that code completion would work.
